testing/acceptance_test.py

```diff
@@ -803,6 +804,40 @@ def test_func(tmpdir):
         result = testdir.runpytest(n)
         result.stdout.fnmatch_lines(["*UserWarning*foo.txt*", "*1 passed, 1 warnings*"])
 
+    @pytest.mark.parametrize("n", ["-n0", "-n1"])
+    def test_unserializable_warning_details(self, testdir, n):
+        """Check that warnings with unserializable _WARNING_DETAILS are
+        handled correctly (#379).
+        """
+        if sys.version_info[0] < 3:
+            # The issue is only present in Python 3 warnings
+            return
+        testdir.makepyfile(
+            """
+            import warnings, pytest
+            import socket
+            import gc
+            def abuse_socket():
+                s = socket.socket()
+                del s
+
+            # Deliberately provoke a ResourceWarning for an unclosed socket.
+            # The socket itself will end up attached as a value in
+            # _WARNING_DETAIL. We need to test that it is not serialized
+            # (it can't be, so the test will fail if we try to).
+            def test_func(tmpdir):
+                warnings.resetwarnings()
+                warnings.simplefilter('always', ResourceWarning)
```

Comment:
Use `@pytest.mark.filterwarnings('always')` instead; using the mark will only affect this inner test. Calling `warnings.simplefilter` will unfortunately affect the outer Python process running pytest as well.

Reply:
I think I addressed this. Let me know if I misunderstood you.